src/Microsoft.IdentityModel.Tokens/Base64UrlEncoder.cs

```diff
@@ -279,30 +278,21 @@ internal static unsafe byte[] UnsafeDecode(ReadOnlyMemory<char> str)
             }
             else
             {
-                if (decodedLength == str.Length)
+                if (decodedLength == strSpan.Length)
                 {
-                    if (MemoryMarshal.TryGetArray(str, out ArraySegment<char> segment))
-                    {
-                        return Convert.FromBase64CharArray(segment.Array, segment.Offset, segment.Count);
-                    }
-                    else
-                    {
-                        bool gotString = MemoryMarshal.TryGetString(str, out string text, out int start, out int length);
-                        Debug.Assert(gotString, "Expected ReadOnlyMemory to wrap either array or string");
-                        return Convert.FromBase64String(text.Substring(start, length));
-                    }
+                    return Convert.FromBase64CharArray(strSpan.ToArray(), 0, strSpan.Length); // TODO: ToArray results in a heap allocation. Is there a better way?
```

Comment:
perhaps: https://learn.microsoft.com/en-us/dotnet/api/system.convert.tryfrombase64string?view=net-8.0

Reply:
Keeping it as it is as strSpan.ToString() would be just as expensive as strSpan.ToString